value.go

```diff
@@ -690,7 +690,9 @@ func (self Value) export() interface{} {
 				}
 				value := object.get(name).export()
 				t = reflect.TypeOf(value)
-				if state == 0 {
+				if t == nil {
```

Comment:
I've added an extra check for `t` being `nil` (which happens when `value` comes from `undefined`) before constructing the common type result, and this prevents that later crash.